src/common/row_set.h

```diff
@@ -150,24 +151,32 @@ class PartitionBuilder {
     }
   }
 
+  // allocate thread local memory, should be called for each specific task
+  void AllocateForTask(size_t id) {
+    if (mem_blocks_[id].get() == nullptr) {
+      BlockInfo* local_block_ptr = new BlockInfo;
```

Comment:
I prefer to add some checks for memory allocation here

Reply:
done